### ISSUE TITLE
The plugin abruptly breaks Gradle builds in case the pre-release section of the version string is empty

### ISSUE THREAD

**Issue by gciatto-unibo** [DanySK/git-sensitive-semantic-versioning-gradle-plugin#2]
### Problem

Assume the following situation:

```kotlin
gitSemVer {
    minimumVersion.set("0.1.0")
    developmentIdentifier.set("")    // <--- NOTICE THIS
    noTagIdentifier.set("")          // <--- NOTICE THIS
    developmentCounterLength.set(2) 
    version = computeGitSemVer()
}
```
the corresponding version string produced by this plugin is
```
0.1.0-+<commit hash here>
```

The absence of characters between `-` and `+` (i.e. the absence of a *pre-release* section) provokes the build to fail because 
```
Computed version: 0.1.0-+<commit hash here> does not match Semantic Versioning 2.0 requirements.
```

It would be better for the plug-in to simply omit the `-` character in case the pre-release section is empty.

### Possible solution

Consider the `fullVersion` inner function in [class `GitSemVer`](/DanySK/git-sensitive-semantic-versioning-gradle-plugin/blob/master/src/main/kotlin/org/danilopianini/gradle/gitsemver/GitSemVer.kt).
Its last line is 
```kotlin
return "$tag-$devString$distance+$hash"
```
You can consider replacing this line with:
```kotlin
val preRelease = devString + distance
return "$tag${if (preRelease.isNullOrBlank()) "" else "-" + preRelease}+$hash"
```

**Comment by DanySK** [comment]
Version 0.2.3-dev1e-2955172 and above work, producing version: `0.1.0+1626789431775`

**Comment by DanySK** [comment]
Fixed in `0.3.0` stable, which will be released automatically in minutes.
